ouroboros-consensus/bench/PerasCertDB-bench/Main.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE LambdaCase #-}
+
+-- | This module contains benchmarks for Peras chain weight calculation as implemented by
+--   the by the 'Ouroboros.Consensus.Storage.PerasCertDB.API.boostedWeightForFragment'
+--   function.
+--
+--   We benchmark the calculation on a static sequence of chain fragments of increasing
+--   length, ranging from 0 to around 2160, with a sampling rate of 100. The chain fragments
+--   are instantiated with 'TestBlock', and every 300 blocks there is a booster block with
+--   weight 15. All parameters are set in 'benchmarkParams'.
+module Main (main) where
+
+import Data.List (iterate')
+import Data.Map.Strict qualified as Map
+import Numeric.Natural (Natural)
+import Ouroboros.Consensus.Block (PerasWeight (PerasWeight))
+import Ouroboros.Consensus.Storage.PerasCertDB.API
+  ( PerasWeightSnapshot (..)
+  , boostedWeightForFragment
+  )
+import Ouroboros.Network.AnchoredFragment qualified as AF
+import Test.Tasty.Bench
+import Test.Util.TestBlock (TestBlock)
+import Test.Util.TestBlock qualified as TestBlock
+
+data BenchmarkParams = BenchmarkParams
+  {samplingRate :: Natural, k :: Natural, boostedBlockRate :: Natural, boostWeight :: PerasWeight}
+
+benchmarkParams :: BenchmarkParams
+benchmarkParams =
+  BenchmarkParams{samplingRate = 100, k = 2160, boostedBlockRate = 300, boostWeight = PerasWeight 15}
+
+main :: IO ()
+main =
+  Test.Tasty.Bench.defaultMain $ map benchBoostedWeightForFragment inputs
+ where
+  -- NOTE: we do not use the 'env' combinator to set up the test data since
+  --       it requires 'NFData' for 'AF.AnchoredFragment'. While the necessary
+  --       instances could be provided, we do not think is necessary for this
+  --       benchmark, as the input data is rather small.
+  inputs :: [(Natural, (PerasWeightSnapshot TestBlock, AF.AnchoredFragment TestBlock))]
+  inputs =
+    getEveryN (samplingRate benchmarkParams) $
+      take (fromIntegral $ k benchmarkParams) $
+        zip [0 ..] $
+          zip (map uniformWeightSnapshot fragments) fragments
+
+benchBoostedWeightForFragment ::
+  (Natural, (PerasWeightSnapshot TestBlock, AF.AnchoredFragment TestBlock)) -> Benchmark
+benchBoostedWeightForFragment (i, (weightSnapshot, fragment)) =
+  bench ("boostedWeightForFragment of length " <> show i) $
+    whnf (boostedWeightForFragment weightSnapshot) fragment
+
+-- | An infinite list of chain fragments
+fragments :: [AF.AnchoredFragment TestBlock]
+fragments = iterate' addSuccessorBlock genesisFragment
+ where
+  genesisFragment :: AF.AnchoredFragment TestBlock
+  genesisFragment = AF.Empty AF.AnchorGenesis
+
+  addSuccessorBlock :: AF.AnchoredFragment TestBlock -> AF.AnchoredFragment TestBlock
+  addSuccessorBlock = \case
+    AF.Empty _ -> (AF.Empty AF.AnchorGenesis) AF.:> (TestBlock.firstBlock 0)
+    (xs AF.:> x) -> (xs AF.:> x) AF.:> TestBlock.successorBlock x
+
+-- | Given a chain fragment, construct a weight snapshot where there's a boosted block every 300 slots
```

Comment:
Currently, this benchmark creates chains where every slot has a block, so every 300th block is boosted.

With the (draft) Peras parameters for mainnet, we would have a Peras boost block every 90 slots (if everything goes well), so due to `f=1/20` we would expect every 4-5 blocks under nominal chain density.